pulp_container/app/registry_api.py

```diff
@@ -392,10 +392,19 @@ def get_dr_push(self, request, path, create=False):
 
     def create_dr(self, path, request):
         with transaction.atomic():
-            repository = serializers.ContainerPushRepositorySerializer.get_or_create({"name": path})
-            distribution = serializers.ContainerDistributionSerializer.get_or_create(
-                {"base_path": path, "name": path}, {"repository": get_url(repository)}
-            )
+            try:
+                repository = serializers.ContainerPushRepositorySerializer.get_or_create(
+                    {"name": path}
+                )
+            except models.ContainerPushRepository.DoesNotExist:
+                raise RepositoryInvalid(name=path, message="Repository is read-only.")
+
+            try:
+                distribution = serializers.ContainerDistributionSerializer.get_or_create(
+                    {"base_path": path, "name": path}, {"repository": get_url(repository)}
+                )
+            except models.ContainerDistribution.DoesNotExist:
+                raise InvalidRequest(message="ContainerDistribution does not exist.")
```

Comment:
I guess we can raise again RepositoryInvalid. But, I have no idea what to tell to client...